projects.podinfo.cue

```diff
@@ -52,23 +56,27 @@ let PODINFO = #KargoProjectBuilder & {
 }
 
 // Register podinfo as a Holos Project.
-Projects: podinfo: PODINFO.Project
+Projects: podinfo: PODINFO.Project.HolosProject
 
 // Register podinfo as a Kargo Project.
-KargoProjects: podinfo: PODINFO.KargoProject
+KargoProjects: podinfo: PODINFO.Project
 
-// Compose stage specific httproutes with the platform.
-HTTPRoutes: PODINFO.Project.httpRoutes
+// Compose stage specific httproutes with the platform selecting namespaces.
+for NS in PODINFO.Project.HolosProject.namespaces {
+	for K, V in NS.metadata.labels {
+		if K == HTTPROUTE_LABEL && V == "podinfo" {
+			HTTPRoutes: (NS.metadata.name): _backendRefs: podinfo: {
+				namespace: NS.metadata.name
+				port:      9898
+			}
+		}
 
-// Manage a backend ref for all prod tier stages.
-HTTPRoutes: podinfo: _backendRefs: {
-	for COMPONENT in PODINFO.Project.components {
-		if COMPONENT._stage != _|_ {
-			if COMPONENT._stage.tier == "prod" {
-				// The field name just needs to be unique, we don't output it.
-				(COMPONENT._namespace): PODINFO.BackendRefs.podinfo.podinfo & {
-					namespace: COMPONENT._namespace
-				}
+		// Manage a backend ref for all prod tier stages.
+		if K == "holos.run/stage.tier" && V == "prod" {
+			HTTPRoutes: podinfo: _backendRefs: (NS.metadata.name): {
+				name:      "podinfo"
+				namespace: NS.metadata.name
+				port:      9898
```

Comment:
Kind of a nice pattern, pass labels into a builder, select against them in the built struct.